Abstract field's getter doesn't get get declared in base type

hutcheb opened the pull request:
bug: abstract field's getter doesn't get get declared in base type
feat: fix for using virtual fields as discriminators
bug: integration tests can now be skipped with the -DskipTests switch

hutcheb committed:
bug: abstract fields don't get declared in base type.
feat: fix for using virtual fields as discriminators
bug: integration tests can now be skipped with the -DskipTests switch

sruehl commented on build-utils/language-java/src/main/resources/templates/java/pojo-template.java.ftlh:
```diff
@@ -23,7 +23,7 @@
 <#-- @ftlvariable name="outputFlavor" type="java.lang.String" -->
 <#-- @ftlvariable name="helper" type="org.apache.plc4x.language.java.JavaLanguageTemplateHelper" -->
-<#-- @ftlvariable name="type" type="org.apache.plc4x.plugins.codegenerator.types.definitions.ComplexTypeDefinition" -->
+<#-- @ftlvariable name="type" type="org.apache.plc4x.plugins.codegenerator.language.mspec.model.definitions.DefaultComplexTypeDefinition" -->
```
```suggestion
<#-- @ftlvariable name="type" type="org.apache.plc4x.plugins.codegenerator.types.definitions.ComplexTypeDefinition" -->```

sruehl replied:
we could add the method to the base interface

sruehl commented on build-utils/protocol-base-mspec/src/main/java/org/apache/plc4x/plugins/codegenerator/language/mspec/model/definitions/DefaultComplexTypeDefinition.java:
```diff
@@ -65,6 +65,11 @@ public boolean isAbstract() {
             .collect(Collectors.toList());
     }
 
+    public List<AbstractField> getAbstractFields() {
```
we should add this method here to `org.apache.plc4x.plugins.codegenerator.types.definitions.ComplexTypeDefinition` and add a `Override` annotation
```suggestion
    @Override
    public List<AbstractField> getAbstractFields() {
```

hutcheb committed:
fix: reverted ComplexTypeDefinition change on tmeplate

fix: Virtual fields getters weren't generated.

fix: String casts now use valueOf for virtual fields within template.

feat: Add tests for virtual fields.

sruehl committed:
Merge branch 'develop' into feat/abstract_fields
